optimization.py

```diff
@@ -130,10 +130,11 @@ def mutate(vec):
     # 1. step not in range check
```

Comment:
I give you some photo fails without the change






also you can see the reason why :

because my change can prevent the negative number raising.




here some photo isn't out of range with my change:



The end. Thanks


 
 
------------------ Original ------------------
From:  "Nico Weber"<notifications@github.com>;
Date:  Wed, Jul 29, 2015 07:12 AM
To:  "nico/collectiveintelligence-book"<collectiveintelligence-book@noreply.github.com>; 
Cc:  "Dylan"<chenzl25@mail2.sysu.edu.cn>; 
Subject:  Re: [collectiveintelligence-book] Test branch (#1)

 

In optimization.py:
 > @@ -130,10 +130,11 @@ def mutate(vec): >      # 1. step not in range check  
Thanks! I think the XXX: broken. in line 129 means that I think that this function is generally wrong. Note that the code is from http://www.amazon.com/Programming-Collective-Intelligence-Building-Applications/dp/0596529325 , I didn't write it myself.
 
Can you explain what problem you are fixing? Maybe you could add a test to optimization_test.py that fails without this change but passes with it?
 
—
Reply to this email directly or view it on GitHub.